### ISSUE TITLE
registry.npmjs.com/<pkg_name>/<version> response uses http not https for tarball URL

### ISSUE THREAD

**Issue by ashleygwilliams** [npm/registry-issue-archive#109]
if you go to `/<pkg_name>` and look at the response, the tarball URLs use https.
if you go to `/<pkg_name>/<version>`, the response contains a tarball URL that uses http.

the expected behavior is that both should use https, and therefore this seems like a bug.

thanks to @davidroberts63 for spotting this!

**Comment by othiym23** [comment]
From npm's Slack:

> [13:44]  no extant version of the CLI uses that data
> [13:44]  there was a long in-air conversation about why we still support `/package/version` endpoints, and the tl;dr is that we tried to remove them when we rolled out registry v2 and people complained because they had tooling or something relying on them
> [13:45]  I don't think any 1.x versions of npm user `/pkg/version`, and I know that nothing `npm@2+` uses them
> [13:46]  the fix is to extend the rewrite rules for the regular package documents to the `/package/version` documents
> [13:46]  _or_ remove `/package/version` responses, which is probably something we could do as part of registry v3, assuming we have some other way to request the package documents for specific versions

**Comment by davidroberts63** [comment]
Yeah, I ran into this while using [ProGet](https://inedo.com/proget). Apparently (via Fiddler) it uses the /package/version form. I'm going to be asking them (Proget) about updates to use /package/version. But I imagine other 3rd parties may be running into it.
